src/modules/land_detector/FixedwingLandDetector.cpp

```diff
@@ -83,7 +83,9 @@ bool FixedwingLandDetector::_get_landed_state()
 			_velocity_z_filtered = val;
 		}
 
-		_airspeed_filtered = 0.95f * _airspeed_filtered + 0.05f * _airspeed.true_airspeed_m_s;
+		// set _airspeed_filtered to 0 if airspeed data is invalid
+		_airspeed_filtered = PX4_ISFINITE(_airspeed_validated.true_airspeed_m_s) ? 0.95f * _airspeed_filtered + 0.05f *
```

Comment:
Suggestion - a simple conditional is usually easier to read than everything packed into one line with the ternary operator.

You could also add a check for airspeed_validated validity (timeout) to handle the airspeed selector failing or even being stopped manually.

Reply:
done.